src/components/SiteHeader.js

```diff
@@ -30,125 +36,158 @@ class SiteHeader extends Component {
+                        icon: <Icon type="user" />,
+                        text: this.props.user.preferred_username,
+                        children: [
+                            {
+                                key: "sign-out-link",
+                                onClick: () => (window.location.href = withBasePath(SIGN_OUT_URL)),
+                                icon: <Icon type="logout" />,
+                                text: <span className="nav-text">Sign Out</span>,
+                            },
+                        ],
+                    },
+                ]
+                : [
+                    {
+                        key: "sign-in",
+                        style: { float: "right" },
+                        icon: <Icon type="login" />,
+                        text: (
+                <span className="nav-text">{this.props.authHasAttempted ? "Sign In" : "Loading..."}</span>
+                        ),
+                        onClick: () => (window.location.href = signInURLWithRedirect()),
+                    },
+                ]),
             {
                 url: withBasePath("notifications"),
-                style: {float: "right"},
+                style: { float: "right" },
                 disabled: !this.props.isOwner,
-                icon: <Badge dot count={this.props.unreadNotifications.length}>
-                    <Icon type="bell" style={{marginRight: "0"}}/>
-                </Badge>,
-                text: <span className="nav-text" style={{marginLeft: "10px"}}>
-                    Notifications
-                    {this.props.unreadNotifications.length > 0 ? (
-                        <span> ({this.props.unreadNotifications.length})</span>
-                    ) : null}
-                </span>,
+                icon: (
+          <Badge dot count={this.props.unreadNotifications.length}>
+            <Icon type="bell" style={{ marginRight: "0" }} />
+          </Badge>
+                ),
+                text: (
+          <span className="nav-text" style={{ marginLeft: "10px" }}>
+            Notifications
+            {this.props.unreadNotifications.length > 0 ? (
+              <span> ({this.props.unreadNotifications.length})</span>
+            ) : null}
+          </span>
+                ),
                 onClick: () => this.props.showNotificationDrawer(),
-            }
+                key: "notifications",
+            },
+            {
+                style: { float: "right" },
+                icon: <Icon type="setting" />,
+                text: <span className="nav-text">Settings</span>,
+                onClick: this.toggleModalVisibility,                //xxxxxxxxx
+                key: "settings",
+            },
         ];
 
         return (
-          <Layout.Header>
-            <Link to={BASE_PATH}>
-              <div
-                style={{
-                    margin: "0 15px 0 0",
-                    float: "left",
-                }}
+            <>
```

Comment:
I think this should be less indented as it wraps the markup that follows.

Reply:
This, and all the other indentation issues, are a product of the linter settings. I agree they don't look great, but they're beyond the scope of this pr (and the pr won't even merge if the linter disagrees with you on the indentation)